drivers/sensor/adxl345/adxl345.c

```diff
@@ -223,66 +323,114 @@ static int adxl345_init(const struct device *dev)
 	}
 
 	rc = adxl345_reg_read_byte(dev, ADXL345_DEVICE_ID_REG, &dev_id);
-	if (rc < 0 || dev_id != ADXL345_PART_ID) {
+	if (rc < 0 || dev_id != ADXL345_DEVICE_ID_VALUE) {
 		LOG_ERR("Read PART ID failed: 0x%x\n", rc);
 		return -ENODEV;
 	}
 
-	rc = adxl345_reg_write_byte(dev, ADXL345_FIFO_CTL_REG, ADXL345_FIFO_STREAM_MODE);
+	data->fifo_mode = adxl345_map_fifo(cfg->fifo_mode);
+	rc = adxl345_reg_write_byte(dev, ADXL345_FIFO_CTL_REG, data->fifo_mode);
 	if (rc < 0) {
 		LOG_ERR("FIFO enable failed\n");
 		return -EIO;
 	}
 
-	rc = adxl345_reg_write_byte(dev, ADXL345_DATA_FORMAT_REG, ADXL345_RANGE_16G);
+	data->range = adxl345_map_range(cfg->range);
+	rc = adxl345_set_range(dev, data->range);
 	if (rc < 0) {
-		LOG_ERR("Data format set failed\n");
 		return -EIO;
 	}
 
-	rc = adxl345_reg_write_byte(dev, ADXL345_RATE_REG, ADXL345_RATE_25HZ);
+	data->odr = adxl345_map_odr(cfg->odr);
+	rc = adxl345_set_output_rate(dev, data->odr);
 	if (rc < 0) {
-		LOG_ERR("Rate setting failed\n");
-		return -EIO;
+		return rc;
 	}
 
-	rc = adxl345_reg_write_byte(dev, ADXL345_POWER_CTL_REG, ADXL345_ENABLE_MEASURE_BIT);
+	rc = adxl345_reg_write_byte(dev, ADXL345_POWER_CTL_REG, ADXL345_POWER_CTL_REG_MEASURE_BIT);
 	if (rc < 0) {
 		LOG_ERR("Enable measure bit failed\n");
 		return -EIO;
 	}
 
+#ifdef CONFIG_ADXL345_TRIGGER
+	rc = adxl345_init_interrupt(dev);
+	if (rc != 0) {
+		LOG_ERR("Failed to initialize interrupt!");
+		return -EIO;
+	}
+#endif
+
 	return 0;
 }
 
-#define ADXL345_CONFIG_SPI(inst)                                       \
-	{                                                              \
-		.bus = {.spi = SPI_DT_SPEC_INST_GET(inst,              \
-						    SPI_WORD_SET(8) |  \
-						    SPI_TRANSFER_MSB | \
-						    SPI_MODE_CPOL |    \
-						    SPI_MODE_CPHA,     \
-						    0)},               \
-		.bus_is_ready = adxl345_bus_is_ready_spi,              \
-		.reg_access = adxl345_reg_access_spi,                  \
+static int adxl345_attr_set(const struct device *dev, enum sensor_channel chan,
+			    enum sensor_attribute attr, const struct sensor_value *val)
+{
+	switch (attr) {
+	case SENSOR_ATTR_WATERFALL_LEVEL:
+		return adxl345_reg_write_mask(dev, ADXL345_FIFO_CTL_REG,
+					      ADXL345_FIFO_CTL_SAMPLE_MSK, 10);
+	case SENSOR_ATTR_ACTIVE_THRESH:
+		return adxl345_reg_write_byte(dev, ADXL345_THRESH_ACT_REG, (val->val1) / 62.5);
```

Comment:
Please address this.

Reply:
Should be already corrected to avoid floating point division: 
https://github.com/whati001/zephyr/blob/7b4fb77898c014e53236b9b0c28f3538f80576f1/drivers/sensor/adxl345/adxl345.c#L383C64-L383C64